tools/concepts/mfcc/test_mfcc.m

```diff
@@ -0,0 +1,219 @@
+% 6. The ceptral coefficiens are output at FFT hop rate. If SOF period is shorter
+%    than hop, then every copy() does not trigger FFT. If period is longer than
+%    hop several FFTs may be computed in a single copy().
+%
+% Some ideas are from Python code example of MFCC in
+% https://www.kaggle.com/code/ilyamich/mfcc-implementation-and-tutorial/notebook
+
+% SPDX-License-Identifier: BSD-3-Clause
+%
+% Copyright (c) 2022, Intel Corporation. All rights reserved.
+
+function test_mfcc()
+
+	%% Get input signal
+	stream.fs = 48e3;
+	stream.t = 1e-3;
+	stream.channels = 1;
+	stream.frames = round(stream.fs * stream.t);
+
+	%% mfcc parameters
+	mfcc.window = 'hanning';
+	mfcc.fft_size = 1024;
+	mfcc.hop_size = 480; % number of samples between successive FFT
+	mfcc.mel_bands = 10;
+
+	%% mfcc state init
+	state = init_mfcc(mfcc, stream);
+
+	input_t = 1.0;
+	t = (0:floor(input_t * stream.fs) - 1)/stream.fs;
+	f0 = 10;
+	f1 = 20e3;
+	x = chirp(t, f0, t(end), f1, 'linear', 90);
+
+	input_frames = (length(x) - stream.frames + 1);
+	all_mel_db = [];
+	all_cc = [];
+	idx = 1;
+	for i = 1:stream.frames:input_frames
+		stream.period = x(i:i + stream.frames - 1);
+		[mel_db, cc, state] = mfcc_copy(stream, mfcc, state);
+
+		% Check if returned one of more Mel spectras
+		if ~isempty(mel_db)
+			s = size(mel_db);
+			for j = 1:s(2)
+				all_mel_db(:,idx) = mel_db;
+				all_cc(:,idx) = cc;
+				idx = idx + 1;
+			end
+		end
+	end
+
+	figure(1)
+	s = size(all_mel_db);
+	b = 1:s(1);
+	n = (0:s(2)-1) * mfcc.hop_size / stream.fs;
+	surf(n, b, all_mel_db, 'EdgeColor', 'none');
+	colormap(jet); view(0,90);
+	xlabel('Time (s)');
+	ylabel('Mel band')
+
+	figure(2)
+	surf(n, b, all_cc, 'EdgeColor', 'none');
+	colormap(jet); view(0,90);
+	xlabel('Time (s)');
+	ylabel('Mel band cepstral coefficients')
+
+end
+
+function [mel_db_matrix, cc_matrix, state] = mfcc_copy(stream, param, state)
+
+	% Copy from stream period to buffer
+	i1 = state.buffer_pos;
+	i2 = i1 + stream.frames - 1;
+	if i2 > state.buffer_size
+		error('Buffer overflow');
+	end
+	state.buffer(i1:i2) = stream.period;
+	state.buffer_pos = i2 + 1;
+	n = i2;
+
+	mel_db_matrix = [];
+	cc_matrix = [];
+	if n > param.hop_size
+		% How many hops
+		m = floor(n / param.hop_size);
+		for i = 1:m
+			% Data for fft_size is available
+			fft_buf = zeros(param.fft_size, 1);
+			fft_buf(1:state.prev_size) = state.prev;
+			i3 = state.prev_size + 1;
+			i4 = param.fft_size;
+			i5 = (i - 1) * param.hop_size + 1;
+			i6 = i5 + param.hop_size - 1;
+			fft_buf(i3:i4) = state.buffer(i5:i6);
+		        %figure(1); plot(fft_buf); grid on;
+
+			% FFT -> power spectra
+			lin_p = do_win_fft(state, fft_buf);
+			%figure(2); plot(lin_p); grid on;
+
+			% Mel bands -> dB -> DCT
+			mel_db = convert_lin_to_mel_db(state, param, lin_p);
+			%figure(3); plot(mel_db); grid on;
+			mel_db_matrix(:,i) = mel_db;
+			cc = dct(mel_db); % DCT type-II (default)
+			%cc = dct(mel_db, 'Type', 3); % DCT type-III
```

Comment:
DCT type to use is not clear. We could assume one or offer commonly used alternatives via parameters.